### PR TITLE
[WIP] [test] Sagemaker HF tests

### PR DESCRIPTION
**Pull request by mansimane**

### CODE REVIEW

File: huggingface/pytorch/training/docker/1.6/py3/cu110/Dockerfile.gpu [hunk 1]
```diff
@@ -14,8 +14,7 @@ RUN pip install --no-cache-dir \
 	datasets==${DATASETS_VERSION}
 RUN apt-get update \
  && apt install git-lfs \
-   	unzip \
- && apt-get clean \  
+ && apt-get clean \
```
Comment (saimidu) [inline huggingface/pytorch/training/docker/1.6/py3/cu110/Dockerfile.gpu:17]: Why remove `unzip` here?

Comment (mansimane) [reply huggingface/pytorch/training/docker/1.6/py3/cu110/Dockerfile.gpu:17]: @SergTogul told that in newly released pytorch base images has `unzip`.  With `unzip` here, pytorch build fails.

File: src/config/build_config.py [hunk 2]
```diff
@@ -5,7 +5,7 @@
 # Do remember to revert it back to False before merging any PR (including NEURON dedicated PR)
 ENABLE_NEURON_MODE = False
 # Frameworks for which you want to disable both builds and tests
-DISABLE_FRAMEWORK_TESTS = []
+DISABLE_FRAMEWORK_TESTS = ["tensorflow", "mxnet", "pytorch"]
```
Comment (saimidu) [inline src/config/build_config.py:8]: To be reverted once PR is approved by reviewer.

File: test/sagemaker_tests/huggingface_tensorflow/training/integration/local/test_placeholder.py [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import pytest
+import os
+
+@pytest.mark.processor("gpu")
+@pytest.mark.model("placeholder")
+@pytest.mark.skip_cpu
+@pytest.mark.skip_py2_containers
+def test_placeholder():
+    pass
```
Comment (saimidu) [inline test/sagemaker_tests/huggingface_tensorflow/training/integration/local/test_placeholder.py:9]: nit: Add newline at the end of all scripts, as some tools can fail when reading python scripts that do not have an empty line at the end.

https://stackoverflow.com/questions/2287967/why-is-it-recommended-to-have-empty-line-in-the-end-of-a-source-file

File: test/sagemaker_tests/huggingface_tensorflow/training/integration/sagemaker/test_smdataparallel.py [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+        role='SageMakerRole',
+        instance_type=instance_type,
+        instance_count=instance_count,
+        image_uri=ecr_image,
+        framework_version=framework_version,
+        py_version=py_version,
+        distribution=distribution,
+        sagemaker_session=sagemaker_session,
+        hyperparameters=hyperparameters,
+        debugger_hook_config=False,  # currently needed
```
Comment (saimidu) [inline test/sagemaker_tests/huggingface_tensorflow/training/integration/sagemaker/test_smdataparallel.py:69]: Please create a GitHub Issue here to track the fact that this flag needs to be removed whenever possible, if it isn't possible to delete this right now.

Also, please add a `#TODO: ` statement around this section of code to let any readers know that this needs to be removed. Maybe you could create the Issue first, and link to that in the TODO statement.

File: test/sagemaker_tests/huggingface_tensorflow/training/integration/sagemaker/test_smdataparallel.py [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+                           role='SageMakerRole',
+                           instance_type=instance_type,
+                           instance_count=instance_count,
+                           image_uri=ecr_image,
+                           framework_version=framework_version,
+                           py_version=py_version,
+                           sagemaker_session=sagemaker_session,
+                           hyperparameters=hyperparameters,
+                           distribution=distribution,
+                           debugger_hook_config=False,  # currently needed
```
Comment (saimidu) [inline test/sagemaker_tests/huggingface_tensorflow/training/integration/sagemaker/test_smdataparallel.py:108]: Please create a GitHub Issue here to track the fact that this flag needs to be removed whenever possible, if it isn't possible to delete this right now.

Also, please add a `#TODO: ` statement around this section of code to let any readers know that this needs to be removed. Maybe you could create the Issue first, and link to that in the TODO statement.

File: test/test_utils/sagemaker.py [hunk 6]
```diff
@@ -196,11 +196,14 @@ def install_sm_local_dependencies(framework, job_type, image, ec2_conn):
         # TF inference test fail if run as soon as instance boots, even after health check pass. rootcause:
         # sockets?/nginx startup?/?
         install_custom_python("3.6", ec2_conn)
+    if framework == "huggingface_tensorflow":
+        install_custom_python("3.7", ec2_conn)
```
Comment (saimidu) [inline test/test_utils/sagemaker.py:200]: This statement doesn't link python3 to python3.7. This is a fix that needs to go into the `install_custom_python` function.

File: test/sagemaker_tests/huggingface_tensorflow/training/integration/local/test_placeholder.py (hunk 3 above)
Comment (mansimane) [reply test/sagemaker_tests/huggingface_tensorflow/training/integration/local/test_placeholder.py]: Thanks, done!

File: test/sagemaker_tests/huggingface_tensorflow/training/integration/sagemaker/test_smdataparallel.py (hunk 4 above)
Comment (mansimane) [reply test/sagemaker_tests/huggingface_tensorflow/training/integration/sagemaker/test_smdataparallel.py:71]: Done

**Pull request closed without merging**